tests/anoph/test_pca.py

```diff
@@ -88,9 +88,14 @@ def test_pca_plotting(fixture, api: AnophelesPca):
     # PCA parameters.
     n_samples = ds.sizes["samples"]
     n_snps_available = ds.sizes["variants"]
-    n_snps = random.randint(n_samples, n_snps_available)
     n_components = random.randint(3, n_samples)
 
+    # If there are more samples than SNPs available, use all SNPs.
+    if n_samples > n_snps_available:
+        n_snps = n_snps_available
+    else:
+        n_snps = random.randint(n_samples, n_snps_available)
+
```

Comment:
I don't think we properly considered the case where the number of SNPs could be very small, less than the number of samples.

I would guess this is getting triggered because the simulated data has a certain degree of missingness, which means that under the default parameter value of 0 for `max_missing_an` then very few SNPs meet that requirement.

One option here could be to set a less strict value of `max_missing_an` for the purposes of this test. 

You could also keep the changes you've made here, but you might also need to be careful about handling the number of components (`n_components`). In the extreme case of very small number of SNPs or samples, the total number of components available in the output of the PCA will be `min(n_samples, n_snps)`. So, e.g., even if you requested `n_components = 4`, if there are only 3 SNPs then you will only get 3 components in the output.

Perhaps this could all be simplified to:

```
    n_snps = random.randint(1, n_snps_available)
    n_components = random.randint(2, min(n_samples, n_snps))
```

...which should be pretty robust to running with a small number of SNPs.

Reply:
Sounds right. Thanks @alimanfoo .